parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java

```diff
@@ -76,6 +81,9 @@ public ParquetMetricsRowGroupFilter(Schema schema, Expression unbound, boolean c
    * @return false if the file cannot contain rows that match the expression, true otherwise.
    */
   public boolean shouldRead(MessageType fileSchema, BlockMetaData rowGroup) {
+    StructType struct = schema.asStruct();
+    this.expr = Binder.bind(struct, Expressions.rewriteNot(expr), caseSensitive);
```

Comment:
Please revert unnecessary changes and move expression binding back into the constructor.

Reply:
Sure.